addons/ui/fnc_openLobbyManager.sqf

```diff
@@ -0,0 +1,400 @@
+
+    private _display = ctrlParent _ctrlButtonExpand;
+    private _ctrlSlots = _display displayCtrl IDC_LM_SLOTS;
+
+    tvExpandAll _ctrlSlots;
+}];
+
+private _ctrlButtonCollapse = _display displayCtrl IDC_LM_COLLAPSE;
+_ctrlButtonCollapse ctrlAddEventHandler ["ButtonClick", {
+    params ["_ctrlButtonCollapse"];
+
+    private _display = ctrlParent _ctrlButtonCollapse;
+    private _ctrlSlots = _display displayCtrl IDC_LM_SLOTS;
+
+    private _fnc_collapse = {
+        // Collapsing [] path causes tree to disappear
+        if !(_this isEqualTo []) then {
+            _ctrlSlots tvCollapse _this;
+        };
+
+        for "_i" from 0 to ((_ctrlSlots tvCount _this) - 1) do {
+            _this + [_i] call _fnc_collapse;
+        };
+    };
+
+    [] call _fnc_collapse;
+}];
+
+private _ctrlButtonOK = _display displayCtrl IDC_OK;
+_ctrlButtonOK ctrlAddEventHandler ["ButtonClick", {
+    collect3DENHistory {
+        params ["_ctrlButtonOK"];
+
+        private _display = ctrlParent _ctrlButtonOK;
+        private _ctrlSlots = _display displayCtrl IDC_LM_SLOTS;
+
+        private _groups = [];
+        private _selected = [];
+
+        // set3DENSelected takes flattened array
+        {
+            _selected append _x;
+        } forEach get3DENSelected "";
+
+        for "_sideIndex" from 0 to ((_ctrlSlots tvCount []) - 1) do {
+            for "_groupIndex" from 0 to ((_ctrlSlots tvCount [_sideIndex]) - 1) do {
+                private _groupPath = [_sideIndex, _groupIndex];
+                private _group = _ctrlSlots getVariable (_ctrlSlots tvData _groupPath);
+                _groups pushBack _group;
+
+                private _leader = leader _group;
+                private _multipleUnits = count units _group > 1;
+
+                private _units = [];
+
+                for "_unitIndex" from 0 to ((_ctrlSlots tvCount _groupPath) - 1) do {
+                    private _unitPath = [_sideIndex, _groupIndex, _unitIndex];
+                    private _unit = _ctrlSlots getVariable (_ctrlSlots tvData _unitPath);
+                    private _description = _unit getVariable QGVAR(description);
+
+                    set3DENSelected [_unit];
+
+                    if (_multipleUnits) then {
+                        do3DENAction "CutUnit";
+                        do3DENAction "PasteUnitOrig";
+                    };
+
+                    _unit = get3DENSelected "Object" select 0;
+                    _units pushBack _unit;
+
+                    add3DENConnection ["Group", _units, _group];
+                    _group selectLeader _leader;
+
+                    _unit set3DENAttribute ["description", _description];
+                };
+            };
+        };
+
+        // Need frame delaying to avoid duplication bugs
+        [_groups, _selected] spawn {
+            params ["_groups", "_selected"];
+
+            private _total = count _groups;
+            startLoadingScreen [localize LSTRING(AdjustingGroupOrder)];
```

Comment:
You have to be extra super careful that you don't get any undefined variables or errors in your code below, else players might get stuck in endless loading screen.